src/Features/Onboarding.php

```diff
@@ -630,8 +630,23 @@ public static function delete_themes_transient() {
 	 *
 	 * @return bool
 	 */
-	protected function is_home_or_setup_wizard_page() {
-		$allowed_paths = array( 'wc-admin', 'wc-admin&path=/setup-wizard' );
+	protected function is_wc_pages() {
+		$allowed_paths = array(
```

Comment:
I think having an exhaustive list like this is a bit problematic since we can't handle navigations from 3PDs whom built & register their pages via WCAdmin. An example of this is WCPay, I've tried and still able to reproduce the issue by navigating from `/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits`.

I don't know the whole context for the usage of the function, but is using regex possible for our case? Or maybe generate the list using our page registry (I don't know if this is possible though)?

Reply:
Thank you @ilyasfoo. That's a really good point. I thought about the 3PDs use cases and I think this change is unrelated after all. This bug only happens if you're on a React router page then navigate to WooCommerce -> Home without a full refresh. Since 3PDs are not in our React router system, users will be forced to refresh when they navigate to WooCommerce -> Home.

Reply:
@moon0326 If I understand this correctly, WCPay registers their pages via our React router system, as seen [here](https://github.com/Automattic/woocommerce-payments/blob/develop/client/index.js#L28-L31) via a filter `woocommerce_admin_pages_list`. This means that any navigation between WCPay pages registered here to WCAdmin pages will not cause a refresh and reproduce the issue.

Reply:
@ilyasfoo that's good to know. Thank you for letting me know 🙏 . I will test it out 👍

Reply:
@ilyasfoo updated in https://github.com/woocommerce/woocommerce-admin/pull/7109/commits/a35bf6feecf464680fc8537e28848746e2b843b1

It now checks to see if the path starts with `wc-admin`. This should catch any pages that are registered under `wc-admin` path.